docs/prompting/thought_generation/chain_of_thought_few_shot/prompt_mining.md

```diff
@@ -1,7 +1,139 @@
+
+These are some examples of mined completions that were provided in the paper
+
+| Manual Prompts                      | Mined Prompts           |
+| ----------------------------------- | ----------------------- |
+| x is affiliated with the y religion | x who converted to y    |
+| The headquarter of x is in y        | x is based in y         |
+| x died in y                         | x died at his home in y |
+| x is represented by music label y   | x recorded for y        |
+| x is a subclass of y                | x is a type of y        |
+
+We can implement back translation using `instructor` as seen below.
+
+```python hl_lines="20-25"
+import instructor
+from openai import AsyncOpenAI
+from pydantic import BaseModel
+import random
+
+client = instructor.from_openai(AsyncOpenAI())
+
+
+class TranslatedPrompt(BaseModel):
+    translation: str
+
+
+async def translate_prompt(prompt: str, from_language: str, to_language: str):
+    return await client.chat.completions.create(
+        model="gpt-4o",
+        messages=[
+            {
+                "role": "system",
+                "content": f"""
+                You are an expert translation assistant.
+                You are going to be given a prompt and
+                asked to translate it from {from_language}
+                to {to_language}. Paraphrase and use
+                synonyms where possible, especially for
+                the examples.
+                """,
+            },
+            {"role": "user", "content": f"Prompt: {prompt}"},
+        ],
+        response_model=TranslatedPrompt,
+    )
+
+
+async def generate_permutation(prompt: str, language: str) -> str:
+    tranlated_prompt = await translate_prompt(prompt, "english", language)
+    backtranslated_prompt = await translate_prompt(
+        tranlated_prompt.translation, language, "english"
+    )
+    return backtranslated_prompt.translation
+
+
+async def generate_prompts(
+    prompt: str, languages: list[str], permutations: int
+) -> list[str]:
+    coros = [
+        generate_permutation(prompt, random.choice(languages))
+        for _ in range(permutations)
+    ]
+    return await asyncio.gather(*coros)
+
+
+if __name__ == "__main__":
+    import asyncio
+
+    prompt = """
+    You are an expert system that excels at Sentiment
+    Analysis of User Reviews.
+
+    Here are a few examples to refer to:
+
+    1. That was a fantastic experience I had! I'm
+    definitely recommending this to all my friends
+    // Positive
+    2. I think it was a passable evening. I don't think
+    there was anything remarkable or off-putting for me.
+    // Negative
+    3. I'm horrified at the state of affairs in this new
+    restaurant // Negative
+
+    Sentence: This was a fantastic experience!
+    """
+    languages = ["french", "spanish", "chinese"]
+    permutations = 2
+
+    generated_prompts = asyncio.run(generate_prompts(prompt, languages, permutations))
+    for prompt in generated_prompts:
```

Comment:
linting on code blocks, runing the pytest-examples  with --update-examples should fix it